#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/samples/contests/robocup/protos/RobocupSoccerField.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/balls/protos/FifaSoccerBall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/factory/containers/protos/WoodenBox.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.4279862014349415 0.4242586482034974 0.7980178010582413 1.803058292801255
  position 0.8967338477054692 -18.37142943984435 38.86599017658288
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RobocupSoccerField {
}
FifaSoccerBall {
  translation 4.9 0 0.08
  size 1
}
Robot {
  translation 0 0 0.01
  children [
    Pose {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 1
            roughness 1
            metalness 0
          }
          geometry Sphere {
            radius 0.05
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.045 0.025
        rotation 0.13448695438559669 -0.13448695438559669 -0.9817466670176033 1.589217
        children [
          DEF WHEEL Pose {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 1 0
                  roughness 1
                  metalness 0
                }
                geometry Sphere {
                  radius 0.03
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.045 0.025
        rotation -0.06235700094427207 0.06235700094427207 -0.9961040150840031 1.5747
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    Camera {
      translation 0.04 0 0.0915
      children [
        Pose {
          translation -0.04 0 0
          rotation -0.5771968951617173 -0.577656895078166 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0.8 0.5
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.025
              }
            }
          ]
        }
      ]
      fieldOfView 1.0472
      width 256
      height 128
      antiAliasing TRUE
      recognition Recognition {
        frameColor 0.929412 0.831373 0
        frameThickness 3
      }
    }
    TouchSensor {
      translation 0.045 0 0.02
      rotation 1 0 0 3.141593
      children [
        Shape {
          appearance PBRAppearance {
            roughness 0.6980390000000001
            metalness 0
          }
          geometry DEF BUMPER Box {
            size 0.02 0.09 0.02
          }
        }
      ]
      name "bumper"
      boundingObject USE BUMPER
      physics Physics {
        density -1
        mass 0.01
      }
      lookupTable [
        0 0 0
        0.01 1 0
      ]
    }
  ]
  name "MyBot"
  boundingObject Pose {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      DEF BODY Cylinder {
        height 0.08
        radius 0.045
      }
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "my_controller"
}
WoodenBox {
  translation 3.28 0 0.15
  size 0.3 0.6 0.3
}
WoodenBox {
  translation 2.59 2.88 0.15
  rotation 0 0 1 1.5708
  name "wooden box(1)"
  size 0.3 0.6 0.3
}
WoodenBox {
  translation 2.12 -2.4 0.15
  rotation 0 0 1 1.5708
  name "wooden box(3)"
  size 0.3 0.6 0.3
}
WoodenBox {
  translation 0.240005 1.94 0.15
  rotation 0 0 1 -3.1415853071795863
  name "wooden box(2)"
  size 0.3 0.6 0.3
}
WoodenBox {
  translation 0.240023 -1.64 0.15
  rotation 0 0 1 -3.1415853071795863
  name "wooden box(4)"
  size 0.3 0.6 0.3
}
